internal/repository/usergrant/user_grant.go

```diff
@@ -165,17 +168,17 @@ func UserGrantCascadeChangedEventMapper(event eventstore.Event) (eventstore.Even
 
 type UserGrantRemovedEvent struct {
 	eventstore.BaseEvent `json:"-"`
-	userID               string `json:"-"`
-	projectID            string `json:"-"`
-	projectGrantID       string `json:"-"`
+	UserID               string `json:"userId"`
+	ProjectID            string `json:"projectId"`
+	ProjectGrantID       string `json:"grantId"`
```

Comment:
In some cases fields may be empty. For example `ProjectGrantID` might not be populated in all cases. In local testing:

```json
{
  "userId": "286399006996234244",
  "grantId": "",
  "projectId": "301720879870509060"
}
```

Please use `omitempty` tags. 

```suggestion
	UserID               string `json:"userId,omitempty"`
	ProjectID            string `json:"projectId,omitempty"`
	ProjectGrantID       string `json:"grantId,omitempty"`
```

Reply:
Added `emitempty` tags along with `roleKeys` field (would also be useful to know which roles to revoke/disable)

Reply:
> Can you give some explanation why that might happen? I'm more interested in user id, as it must clearly exist when the grant is being removed. How should clients (handle the case / interpret the event) when user id is empty?

I'm not sure. I had a quick look to the code and can't find when `ProjectGrantID` should or should not be populated. I also don;t want to spend too much time on that. The existing events also use `omitemtpy` and I prefer to keep it consistent.

Yes the user ID shouldn't be empty. But in the subject of client handling, be aware that event payload is currently not part of any API contract. A client getting events from Zitadel should be hardened to deal with missing or extra fields in any case. The model of an event may always change in the future.

Reply:
Got it, thanks for the explanation. 

> event payload is currently not part of any API contract

We'll have to adjust our current architechture and plans then. Are there any plans to stabilise the event payload as part of the API?

Reply:
Yes, we are having internal talks but nothing concrete enough to bring it to a public discussion and/or roadmap.